tests/UI/campaigns/functional/BO/10_payment/02_preferences/04_carrierRestrictions.js

```diff
@@ -0,0 +1,153 @@
+  // before and after functions
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+    page = await helper.newTab(browserContext);
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  it('should login in BO', async function () {
+    await loginCommon.loginBO(this, page);
+  });
+
+  it('should go to \'Payment > Preferences\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToPreferencesPage', baseContext);
+
+    await dashboardPage.goToSubMenu(
+      page,
+      dashboardPage.paymentParentLink,
+      dashboardPage.preferencesLink,
+    );
+
+    await preferencesPage.closeSfToolBar(page);
+
+    const pageTitle = await preferencesPage.getPageTitle(page);
+    await expect(pageTitle).to.contains(preferencesPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {action: 'uncheck', paymentModule: 'ps_wirepayment', exist: false}},
+    {args: {action: 'check', paymentModule: 'ps_wirepayment', exist: true}},
+    {args: {action: 'uncheck', paymentModule: 'ps_checkpayment', exist: false}},
+    {args: {action: 'check', paymentModule: 'ps_checkpayment', exist: true}},
+  ];
+
+  tests.forEach((test, index) => {
+    it(`should ${test.args.action} free prestashop carrier for '${test.args.paymentModule}'`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', test.args.action + test.args.paymentModule, baseContext);
+
+      const result = await preferencesPage.setCarrierRestriction(
+        page,
+        0,
+        test.args.paymentModule,
+        test.args.exist,
+      );
+
+      await expect(result).to.contains(preferencesPage.successfulUpdateMessage);
+    });
+
+    it('should go to FO and add the first product to the cart', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        `check_${test.args.paymentModule}_${test.args.exist}`,
+        baseContext,
+      );
+
+      // Click on view my shop
+      page = await preferencesPage.viewMyShop(page);
+
+      // Change language in FO
+      await homePage.changeLanguage(page, 'en');
+
+      // Go to the first product page
+      await homePage.goToProductPage(page, 1);
+
+      // Add the product to the cart
+      await productPage.addProductToTheCart(page);
+
+      // Proceed to checkout the shopping cart
+      await cartPage.clickOnProceedToCheckout(page);
+
+      const isCheckoutPage = await checkoutPage.isCheckoutPage(page);
+      await expect(isCheckoutPage).to.be.true;
+    });
+
+    // Personal information step - Login
+    it('should login and go to address step', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        `loginToFO${index}`,
+        baseContext,
+      );
+
+      if (index === 0) {
+        // Personal information step - Login
+        await checkoutPage.clickOnSignIn(page);
```

Comment:
Could it be possible to log in the customer before going into the loop ?

Reply:
Yes it's possible and it's done